src/ape/exceptions.py

```diff
@@ -217,3 +217,9 @@ def __init__(self, snapshot_id: "SnapshotID"):
             snapshot_id = humanize_hash(snapshot_id)  # type: ignore
 
         super().__init__(f"Unknown snapshot ID '{str(snapshot_id)}'.")
+
+
+class QueryEngineError(ApeException):
+    """
+    Raised when a query engine is unavailable.
```

Comment:
Seems like this doc should be more generic like `Raised when issues occur in a query engine.` That way, we can re-use tis exception in more places if we ever need.